### PR TITLE
[Editor] Cleanup editor component

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #1188 

Opening this PR as a conversation starter. I haven't tested it yet, so it likely would need some tlc before it could land. @jlongster I'm curious to see what you think. Note, we've discussed some of these ideas before but I think the project has evolved sufficiently since then.

### CODE REVIEW

**Comment by jasonLaster** [comment]
Forgot to mention this during standup because it's minor.

File: public/js/utils/editor.js [hunk 1]
```diff
@@ -22,7 +26,154 @@ function shouldShowFooter(selectedSource) {
...
+  const { selectedSource } = this.props;
+
+  if (!selectedSource || !shouldShowFooter(selectedSource.toJS())) {
+    return "100%";
+  }
+
+  return "";
+}
+
+function updatePendingJumpLine(props, prevProps) {
```
Comment (wldcordeiro) [inline public/js/utils/editor.js]: If all the function uses is the `selectedLocation` property on the two, why can't we just pass them in as the signature for this?

File: public/js/utils/editor.js [hunk 2]
```diff
@@ -22,7 +26,154 @@ function shouldShowFooter(selectedSource) {
...
+
+function setDebugLine(editor, selectedFrame, selectedLocation) {
+  if (selectedFrame && selectedLocation &&
+      selectedFrame.location.sourceId === selectedLocation.sourceId) {
+    const line = selectedFrame.location.line;
+    editor.codeMirror.addLineClass(line - 1, "line", "debug-line");
+  }
+}
+
+function editorHeight() {
```
Comment (wldcordeiro) [inline public/js/utils/editor.js]: Relying on the `this` to be available in this function seems messy, `selectedSource` should be an argument imo.

File: public/js/utils/editor.js (hunk 1 above)
Comment (jasonLaster) [inline public/js/utils/editor.js]: that's a good point

File: public/js/utils/editor.js (hunk 2 above)
Comment (jasonLaster) [inline public/js/utils/editor.js]: definitely, that's an oversight. this was just designed to sketch the idea

Comment (wldcordeiro) [inline public/js/utils/editor.js]: Cool, figured it wasn't entirely ready from your comment, but thought I'd give some review to help guide along. 😄

**Comment by jlongster** [comment]
What is the goal here? I don't really understand the desire to split up components into a simple file in `components` and a lot of implementation details in `utils`. Components should be as complex as they need to be. It makes it harder to work with when things are arbitrarily split up.

The code in `utils` has things like `shouldShowFooter` and other things that are UI-specific. If anything, I'd rather be moving things from util into components where we can.

**Comment by jasonLaster** [comment]
> what is the goal

the rationale is in [1188](https://github.com/devtools-html/debugger.html/issues/1188), which is easy to miss. 

The utils are similar to [editor.js](https://dxr.mozilla.org/mozilla-central/source/devtools/client/sourceeditor/editor.js). The goal is to have have an editor api for commands/queries.

One benefit would be in the Atom integration work, it would be possible to re-implement methods like `showSourceText` for atom.

> shouldShowFooter

I agree. This is a bad example, that i wish were in the component aswell. `showMessage`, `showSourceText` are better examples. 

I am sharing early to get your opinion. Happy to close.

**Comment by jasonLaster** [comment]
closed as this was mostly for the conversation

**Pull request closed without merging**